iceoryx2-bb/lock-free/src/spmc/unrestricted_atomic.rs

```diff
@@ -51,6 +51,17 @@ impl<T: Copy> Producer<'_, T> {
     pub fn store(&self, new_value: T) {
         self.atomic.store(new_value);
     }
+
+    #[doc(hidden)]
+    pub unsafe fn get_ptr_to_write_cell(&self) -> *mut T {
+        let write_cell = self.atomic.write_cell.load(Ordering::Relaxed);
+        unsafe { (*self.atomic.data[write_cell as usize % NUMBER_OF_CELLS].get()).as_mut_ptr() }
+    }
+
+    #[doc(hidden)]
+    pub unsafe fn update_write_cell(&self) {
+        self.atomic.write_cell.fetch_add(1, Ordering::Relaxed);
```

Comment:
I think the memory order here must be `Release`. I assume that one first writes the content of the write cell and when the update of the pointer was finished `update_write_cell` is called. When now the other side, the reader, consumes the new memory location the underlying payload needs to be synced - therefore it needs to be `Release`.